app/src/main/java/com/android/sample/model/profile/ProfileViewModel.kt

```diff
@@ -55,6 +55,22 @@ open class ProfileViewModel @Inject constructor(private val repository: Profiles
         onFailure = {})
   }
 
+  fun addLikedActivity(userId: String, activityId: String) {
+    repository.addLikedActivity(
+        userId = userId,
+        activityId = activityId,
+        onSuccess = { fetchUserData(userId) },
+        onFailure = {})
+  }
+
+  fun removeLikedActivity(userId: String, activityId: String) {
+    repository.removeLikedActivity(
+        userId = userId,
+        activityId = activityId,
+        onSuccess = { fetchUserData(userId) },
+        onFailure = {})
```

Comment:
The onFailure callback is currently empty here too, so consider doing the same thing as said higher.